### PR TITLE
Survey data converter implementation (Survey list, Survey, Job)

### LINKED ISSUE #1933: Load survey list using new structure

### LINKED ISSUE #1934: Load survey and jobs using new structure

### PR DESCRIPTION
**Pull request by rfontanarosa**
closes #1933
closes #1934

### CODE REVIEW

File: web/src/app/converters/survey-data-converter.ts [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+        new TaskConditionExpression(
+          TaskConditionExpressionType.ONE_OF_SELECTED,
+          multipleChoice!.taskId!,
+          List(multipleChoice!.optionIds!)
+        ),
+      ])
+    );
+  }
+
+  let multipleChoice = undefined;
```
Comment (gino-m) [inline web/src/app/converters/survey-data-converter.ts:155]: Same here - `multipleChoicePbToModel`?

File: web/src/app/services/data-store/data-store.service.ts [hunk 2]
```diff
@@ -77,43 +78,26 @@ export class DataStoreService {
...
-      );
-  }
+    return combineLatest([
+      this.db.collection(SURVEYS_COLLECTION_NAME).doc(id).valueChanges(),
+      this.db
+        .collection(`${SURVEYS_COLLECTION_NAME}/${id}/jobs`)
+        .valueChanges(),
+    ]).pipe(
+      switchMap(surveyAndJobs => {
+        const [s, jobs] = surveyAndJobs;
```
Comment (gino-m) [inline web/src/app/services/data-store/data-store.service.ts:88]: Please extract into function - `loadJobs$`?

File: web/src/app/services/data-store/data-store.service.ts [hunk 3]
```diff
@@ -77,43 +78,26 @@ export class DataStoreService {
...
-            throw Error(`Job $jobId not found`);
-          }
-
-          return job;
-        })
-      );
+        const survey = surveyDocToModel(
+          id,
+          s as DocumentData,
+          jobs as DocumentData[]
```
Comment (gino-m) [inline web/src/app/services/data-store/data-store.service.ts:93]: This seems like the wrong place to combine the data.. can we jus t do the conversion here and not below?

File: web/src/app/services/data-store/data-store.service.ts [hunk 4]
```diff
@@ -477,24 +464,30 @@ export class DataStoreService {
...
-          const data = doc.data() as DocumentData;
-          const survey = FirebaseDataConverter.toSurvey(surveyId, data);
-          const tasks = survey.getJob(jobId)?.tasks;
-          if (!tasks || typeof tasks === 'undefined') {
-            return List<Task>();
-          }
-          return tasks.toList()!;
-        })
-      );
+    return combineLatest([
```
Comment (gino-m) [inline web/src/app/services/data-store/data-store.service.ts:467]: This method could then be defined by calling `loadJobs$().filter()`

File: web/src/app/converters/survey-data-converter.ts (hunk 1 above)
Comment (rfontanarosa) [reply web/src/app/converters/survey-data-converter.ts]: I've used taskMultipleChoicePbToModel

File: firestore/firestore.rules [hunk 5]
```diff
@@ -157,11 +157,11 @@
     // Apply passlist and survey-level ACLs to job documents.
     match /surveys/{surveyId}/jobs/{jobId} {
       // Allow if user has has read access to the survey.
-      allow read: if canViewSurvey(getSurvey(surveyId));
-      // Allow if user is owner of the new submission and can collect data.
-      allow create: if canManageSurvey(getSurvey(surveyId)) || isCreator(request.resource);
-      // Allow if user is owner of the existing submission or can manage survey.
-      allow write: if canManageSurvey(getSurvey(surveyId)) || isCreator(resource);
+      allow read: if canViewSurvey(getSurvey(surveyId), getEmail());
```
[the remark this answers is not in the document]
Comment (rfontanarosa) [reply firestore/firestore.rules:160]: sorry, there was an inconsistency between the source and cloud firestore (of instance). now they are aligned.

**Review by gino-m — APPROVED** [review]
LGTM!

**Pull request merged**